hack/set-motd/main.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+)
+
+type stringFlag struct {
+	set   bool
+	value string
+}
+
+func (sf *stringFlag) Set(x string) error {
+	sf.value = x
+	sf.set = true
+	return nil
+}
+
+func (sf *stringFlag) String() string {
+	return sf.value
+}
+
+func setMotd(motd *stringFlag, user *stringFlag, pr *stringFlag, customPath *stringFlag) {
+	setHelpMsg := `
+	Usage: set-motd set [-motd string] [-user string] [-pr link]
+	  - motd: set custom message
+	  - user: set user using the server
+	  - pr:	set pull-request being tested
+	  - path: set custom path (Default: /etc/motd)
+	
+	`
+
+	if len(os.Args) == 2 || len(os.Args) > 10 {
+		fmt.Print(setHelpMsg)
+		os.Exit(1)
+	}
+
+	currentTime := time.Now()
+	var motdMsg string = fmt.Sprintf("Updated at %s\n", currentTime.Format(time.UnixDate))
+
+	if motd.set {
+		motdMsg = fmt.Sprintf("%s%s\n", motdMsg, motd.String())
+	}
+
+	if pr.set {
+		motdMsg = fmt.Sprintf("%sPull Request %s test initiated by %s\n", motdMsg, pr.String(), user.String())
+	}
+
+	if user.set && !pr.set {
+		motdMsg = fmt.Sprintf("%sUsed by %s\n", motdMsg, user.String())
+	}
+
+	// Clean file
+	var path string = customPath.String()
+	if err := os.Truncate(path, 0); err != nil {
+		log.Printf("Failed to truncate: %v", err)
+	}
+
+	// Open file
+	var f, err = os.OpenFile(path, os.O_RDWR, 0644)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	defer f.Close()
+
+	// Write motd
+	_, err = f.WriteString(motdMsg)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func unsetMotd() {
+	unsetHelpMsg := `
```

Comment:
should be constant or maybe an array with the previous one

Reply:
Fixed.